designcompose/src/main/java/com/android/designcompose/squoosh/SquooshRoot.kt

```diff
@@ -575,9 +576,15 @@ fun SquooshRoot(
                     // We use a custom layout for children that just passes through the layout
                     // constraints that we calculate in our layout glue code above. This gives
                     // children a way to report their sizes to the Rust layout implementation.
-                    key(child.node.layoutId) {
-                        SquooshChildLayout(modifier = composableChildModifier, child = child)
-                    }
+                    val isListItem =
+                        child.node.parent?.view?.name?.let {
+                            customizationContext.getContent(it) != null
+                        } ?: false
+                    if (isListItem)
```

Comment:
I used `key` to avoid list items getting recomposed when they get reordered, so this still meets that need. I am wondering if we had an `AndroidView` in a list item if we'd just see the same thing?

I suspect the crash we're seeing in the DriverUI demo app is really a race condition for a really short-lived Filament View, and it should just handle getting created and destroyed before it has finished loading.

I think this is OK with an updated comment on why key is used, and why it's not used, but also worth checking if it really is that we're just always crashing with a short lived Filament component (our code around Filament is doing some async stuff which probably is where the problem is).

Reply:
I thought it was related to Filament at first, but we tested with `AndroidView` for the backup camera and the map video, and those crashed as well.

Reply:
Oh :(, this seems like a bug higher up then, so this workaround is OK with a comment and link to a ticket.

Reply:
Done